datamodel_code_generator/__main__.py

```diff
@@ -283,6 +283,9 @@ def merge_args(self, args: Namespace) -> None:
             if getattr(args, f) is not None
         }
 
+        if set_args.get('output_model_type') == DataModelType.MsgspecStruct.value:
+            set_args['use_annotated'] = True
```

Comment:
@indrat 
Doesn't msgspec use `Field`? and is it always annotated?
You define some logic for `Field`. But, the unit tests don't cover the parts.
We should change the coverage to 100% before merging.

Reply:
I've added some output tests for when a `msgspec.field` is generated and tried to boost the coverage for the const/alias fields as well.

Regarding forcing`use_annotated` I believe the constraints such as `gt`, `lt`, etc must be attached using `Annotated[..., Meta(gt=..., lt=...)]` etc. So aiming for the least surprise I felt it made sense to always force `use_annotated` on otherwise you won't get the constraints or description metadata generated.

[`msgspec.field`](https://jcristharif.com/msgspec/api.html#msgspec.field) only accepts `default`,`default_factory`, and `name` so it's a little different to `pydantic.Field`